### PR TITLE
Updating lab api implementation to work with new changes

### PR DESCRIPTION
**Pull request by trwalke**
Fixing broken tests by updating lab api implementation.

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1460

### CODE REVIEW

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabUser.cs [hunk 1]
```diff
@@ -39,19 +39,37 @@ public class LabUser
         [JsonProperty("labname")]
         public string LabName { get; set; }
 
-        public FederationProvider FederationProvider { get; set; }
+        //TODO: request the lab to add the federation provider to the lab user response
+        public FederationProvider FederationProvider
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabUser.cs]: @bgavrilMS they seem to have removed the federation provider from the lab response.

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs [hunk 2]
```diff
@@ -54,27 +50,37 @@ private async Task<LabResponse> GetLabResponseFromApiAsync(UserQuery query)
...
 
-            var response = responses[0];
-            response.User.CredentialUrl = response.Lab.CredentialVaultkeyName;
-            response.User.TenantId = response.Lab.TenantId;
-            response.User.FederationProvider = response.Lab.FederationProvider;
+            var user = userResponses[0];
 
-            return response;
+            var appResponse = await GetLabResponseAsync(LabApiConstants.LabAppEndpoint + user.AppId).ConfigureAwait(false);
+            LabApp labApp = JsonConvert.DeserializeObject<LabApp>(StripBracketsFromLabResponse(appResponse));
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs]: @bgavrilMS the new lab app endpoint also adds extra brackets like the user endpoint used to. Ill ask gladwin to remove

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs [hunk 3]
```diff
@@ -54,27 +50,37 @@ private async Task<LabResponse> GetLabResponseFromApiAsync(UserQuery query)
...
+
+            return new LabResponse
+            {
+                User = user,
+                App = labApp
+            };
         }
 
+        private static string StripBracketsFromLabResponse(string response)
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs]: Let's not do this, let's try to deserialize the json. If not possible and the lab will change this reponse, please add a comment.

Comment (trwalke) [reply tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs]: fixed

**Pull request merged**